packages/grid/x-data-grid-pro/src/hooks/features/detailPanel/useGridDetailPanel.ts

```diff
@@ -98,7 +100,10 @@ export const useGridDetailPanel = (
       if (expandedRowIds.length === 0 || !expandedRowIds.includes(row.id)) {
         return { ...initialValue, detail: 0 };
       }
-      const heightCache = gridDetailPanelExpandedRowsHeightCacheSelector(apiRef.current.state);
+      const heightCache = gridDetailPanelExpandedRowsHeightCacheSelector(
+        apiRef.current.state,
+        apiRef.current.instanceId,
+      );
```

Comment:
```suggestion
      const heightCache = gridDetailPanelExpandedRowsHeightCacheSelector(apiRef);
```

Reply:
But shouldn't it work by just passing `apiRef` since it uses our `createSelector` ?